### ISSUE TITLE
[NEW] Open Notarization Protocol

### ISSUE THREAD

**Issue by mehranshakeri** [ethberlinzwei/Find-A-Team#11]
## Introduction

Hi, I'm a software engineer passionate about decentralized systems. I've spent last 3 years digging blockchain related topics. If you need more personal information you can have a look at my [linkedin](https://www.linkedin.com/in/mehran-shakeri-8833b347) however if the idea is more interesting, you can stay in this page :D

-  _Who you are_: Mehran Shakeri
-  _What you do_: Evaluation of Technologies and use cases by building PoCs (proof of concept)
-  _How you got first into Ethereum_: Buying some Ether and then coding Solidity
-  _What brings you to ETHBerlin_: Open Decentralized Future that gives anyone an equal chance

## Idea description

The reality is that not every business can transact directly on the main net. That's why there are side chains, Plasma chains even alternative blockchains. But Ethereum is the most neutral, secure and robust option. For whatever reason that a business might decide to bring its transactions on an alternative rather than Ethereum, it will end up to locking its assets in that system. I have some experience with HL Fabric and Corda which pose this risk. 

_The goal is to define a protocol to enable the users of those systems to unlock their assets while using non-Ethereum solutions._

Let's call the non-Ethereum networks _Local Notary_ (or notary in short) since they are simply notarizing and attesting the latest state of an asset. The protocol starts by registering these notaries in a smart contract in the Ethereum main net (Ethereum in short) and stake some Ether as their security deposit.

Notaries need to submit some commitments in the Ethereum periodically which represents their current state. (e.g. Merkle Root). Users of those systems should receive a signature from the operator on the inclusion of their state in that commitment. This can be done by creating incentive for notaries that is simply paying them more. Later on if the user needs to move her/his asset to another notary (which is another network), s/he should bring that signature and prove her/his asset is part of the latest commitment registered in the Ethereum.

I have to do a bit more investigation, but I think this can also help interoperability among different blockchains by staking some Ether in Ethereum and following this protocol.

I'm working on this topic already for few month and I'm confident it has some potential. (Otherwise you prove me I'm wrong :D ).

## Skillset

_What skills do you need, or think you might need to implement the idea?_

- **Game theory** is super important to challenge the rules of protocol to make sure I've not missed anything critical :D 
- Some background in decentralized thinking/mindset would be helpful

If we go for a MVP then
- Smart Contracts (Solidity, Vyper)
- DApps, Web3.JS (JavaScript)

I'm a programmer myself so I will contribute in the code as well.

## Communication

Riot: @mehranshakeri:matrix.org

If it's interesting for you, either contact me here or linkedin or let me know where you have an account and I will try to join. Otherwise we can meet in person if you are already in Berlin :)

**Comment by step21** [comment]
Sounds interesting. Though is this not kinda like a relay or similar 'interoperability' concepts?

**Comment by mehranshakeri** [comment]
@step21  Well, our main goal is to unlock the assets but I guess it can help interoperability as well. I need more investigation.
Are you interested to have more details? I'm still in TU Berlin. If you are around we can have a chat.

**Comment by step21** [comment]
Hey, I'm pretty sure that exists already. Maybe not as a general thing, but then because depending on what you want to 'notarize' you need to write custom code anyway.

**Comment by mehranshakeri** [comment]
There are Polkadot, Cosmos, POA that I'm aware of. And none of them are as decentralized as Ethereum main net. All of them can lock the assets if they want.
The main difference is they aim for interoperability, that's why they are fine with loosing decentralization but enabling interoperability. On the other hand, I want to give a chance to the real owner of asset/token/state to prove his ownership without any interaction from those validators/node operators or bridges.
E.g. with NFTs in mind, based on my approach, minting or changing the state of a NTF in a (let's call it) side chain is not finalized as far as there is no commitment in the Ethereum main net. And this commitment includes enough information that the new owner of that NFT can claim his right in the main net.
Recently I've started to read about [tBTC](https://tbtc.network) which might be relevant but I have to read the paper first :D

**Comment by mehranshakeri** [comment]
Yes, you can say my definition of decentralization is BitCoin maximalist :D but that's what I need as a neutral entity to rule internationally regardless of border and identity. That's why still I need Ethereum main net not my own blockchain or a small group of people.
